### PR TITLE
docs: update untested iOS keyboard codes

### PR DESCRIPTION
**Pull request by onebigdoor**
update untested iOS keycodes
i'm unsure my method correctly tests consumer codes, so they are excluded
this gist includes the method i used in iOS/xcode to generate my test output, as well as my test output
https://gist.github.com/onebigdoor/5ee09fd5148a925b3b272a1f7435c7b0

attached the keymap i used `keycode.keymap` as `keycode.txt`
[keycode.txt](https://github.com/zmkfirmware/zmk/files/5519529/keycode.txt)

### CODE REVIEW

**Comment by innovaker** [comment]
Thanks @onebigdoor!

I've verified your analysis of the log you provide using a script.  Our results (commits) only differed on two codes:
 - `CAPSLOCK` ([`0x07 0x39`])
   - commit: `true`
   - log: `false`
- `K_POWER` ([`0x07 0x66`])
   - commit: `null`
   - log: `false`

Please could you independently verify the above and update your PR if you agree?

Also, please could you answer the following Qs (it's the PR template I've been working on) ...

```
### iOS version/build?

### What codes did you test?

### How did you test?

### Have you any useful information?

### Has anyone else tested/verified it?
```

Muchas gracias.

**Comment by innovaker** [comment]
I just noticed `K_POWER` wasn't in the keymap.  My bad.  Feel free to test it anyway.

**Comment by onebigdoor** [comment]
oh, got it.  i'll test `K_POWER` and update

**Comment by onebigdoor** [comment]
i manually tested `K_POWER` and confirmed that it worked, so i've updated that.

**Comment by innovaker** [comment]
Thanks for doing this!  For prosperity, please may I check:
- What was the iOS version?
- Was this run on a device, i.e. an iPhone?  Or emulated?  Please could you give a few more details for those of us unfamiliar with iOS development.

**Comment by innovaker** [comment]
For future observers, this was the contents of the original gist:

### Code
```objective-c
// used this method to print all keypresses to console output

override func pressesBegan(_ presses: Set<UIPress>, with event: UIPressesEvent?) {
        super.pressesBegan(presses, with: event)
        guard let key = presses.first?.key else { return }
        var modifiers = ""
        if (key.modifierFlags.contains(.alphaShift)) {
            modifiers += "caps lock "
        }
        if (key.modifierFlags.contains(.shift)) {
            modifiers += "shift "
        }
        if (key.modifierFlags.contains(.command)) {
            modifiers += "command "
        }
        if (key.modifierFlags.contains(.alternate)) {
            modifiers += "alt "
        }
        if (key.modifierFlags.contains(.control)) {
            modifiers += "control "
        }
        if (key.modifierFlags.contains(.numericPad)) {
            modifiers += "numPad "
        }
        

        print(String(format:"%02X", key.keyCode.rawValue), " ", modifiers)
    }
```

### Log
[log.txt](https://github.com/zmkfirmware/zmk/files/5531249/log.txt)

**Comment by onebigdoor** [comment]
tested on physical device, iPhone 7, running iOS 13.6.1 connected to xcode in debug mode.  the xcode project is essentially blank except for the added `pressesBegan` method logging keypresses to console.

**Review by innovaker — APPROVED** [review]
I can confirm that the resulting `diff` matches what I'd expect based on confirmation analysis of the log + comments.

@petejohanson, how would you prefer to play this please?  @onebigdoor's code has effectively run through the Keyboard codes to check for iOS support.  My only concern is whether we could be missing any useful background information/footnotes (i.e. app support), but the same could be said for the existing Linux and Android codes.  Are you happy to merge or should we wait for verification by other iOS testers first?  Thanks!

@petejohanson, whenever we merge, please squash these commits into one and rename appropriately.

**Comment by innovaker** [comment]
Converted to draft as [discussed](https://discordapp.com/channels/719497620560543766/719544230497878116/776473031123795968).  Is on hold for further verification by @onebigdoor.

**Comment by innovaker** [comment]
@petejohanson, changes accepted, though please check https://github.com/zmkfirmware/zmk/pull/356#pullrequestreview-529176430 before merging (if you haven't already seen it).

**Comment by innovaker** [comment]
I'd like to suggest further investigating the `true` options such as `K_MUTE` and `K_PASTE` etc. before merging this PR.  Given what we saw on #377 with `K_POWER`, I'm concerned these could be false positives - i.e. the OS recognises the codes but doesn't act on them.

@onebigdoor, any thoughts on that please?  I wouldn't expect you to do it, we can draw up a checklist for others to verify manually.  We could also consider splitting the PR into those which that are a safe bet vs. those that aren't.

**Comment by onebigdoor** [comment]
i manually tested `CAPSLOCK` and confirmed it works as expected, sending code `39` through the xcode console using same method.  i reran the same test script, and it's still skipping `CAPSLOCK`.  i combed through it, but haven't found the smoking gun yet.

**Comment by innovaker** [comment]
My thoughts on this are the same as my [comment](https://github.com/zmkfirmware/zmk/pull/377#issuecomment-726696654) on macOS.

**Comment by innovaker** [comment]
Please see my [comment](https://github.com/zmkfirmware/zmk/pull/377#issuecomment-727014384) on macOS.

**Comment by onebigdoor** [comment]
updated reverting codes we want to manually test to `null`.  i left were F13-F24 which i'm confident came through correctly, and can be additionally documented further if necessary.  i also kept `CAPSLOCK` and `K_POWER`, and added `C_POWER` because i had tested them manually.

File: docs/src/data/hid.js
Comment (innovaker) [inline docs/src/data/hid.js:2268]: Are you sure this doesn't need manual testing?

File: docs/src/data/hid.js
Comment (innovaker) [inline docs/src/data/hid.js:2243]: Are you sure this doesn't need manual testing?

File: docs/src/data/hid.js
Comment (innovaker) [inline docs/src/data/hid.js:2988]: Are you sure this doesn't need manual testing?

**Review by innovaker — COMMENTED** [review]
Thanks @onebigdoor!  I compared it against the sheet we were using and your feedback.  Only a few follow-up queries.

File: docs/src/data/hid.js
Comment (onebigdoor) [reply docs/src/data/hid.js]: nope.  those three escaped my revision.  i appreciate your thoroughness.  i've updated those three keys to `null`

**Comment by innovaker** [comment]
When doing a final visual check on the rendered page this morning, `K_APPLICATION` caught me eye.  What does it actually do on iOS?  On Windows I believe it's the equivalent of the right-click menu.

**Comment by innovaker** [comment]
Paging @onebigdoor, did you see my last message please?

**Comment by onebigdoor** [comment]
`K_APPLICATION` has no effect on the apps i tried: Gmail, Safari, and Notes.

**Comment by innovaker** [comment]
@onebigdoor, I've rebased this PR as [discussed](https://discord.com/channels/719497620560543766/784523018050469939/786244439563042816), however I couldn't force-push into your repository.

Instead, I've pushed it to: https://github.com/innovaker/zmk/tree/docs/ios-keyboard-codes

Please could you pull it and force-push it to your repository so that this PR updates?

I'd also be grateful if you could please eyeball it for mistakes because every line conflicted (as anticipated).

**Comment by innovaker** [comment]
> `K_APPLICATION` has no effect on the apps i tried: Gmail, Safari, and Notes.

Should this be reflected in the PR?  The deploy preview has a star for `K_APPLICATION` on iOS.

**Comment by onebigdoor** [comment]
@innovaker no.  i don't know why it closed.  i'm working on the push / merge.

**Comment by onebigdoor** [comment]
i think i've got it.  just need to add a footnote about `K_APPLICATION` because the keycode comes through even though it apparently does nothing.  you can reopen, or i will in a bit.

**Comment by innovaker** [comment]
> @innovaker no. i don't know why it closed. i'm working on the push / merge.

ah ok!  I think you might be the only person able to reopen it!  It should be the `Reopen and comment` button next to the `Comment` button :)

**Comment by innovaker** [comment]
> i think i've got it. just need to add a footnote about `K_APPLICATION` because the keycode comes through even though it apparently does nothing. you can reopen, or i will in a bit.

Yeah, I feel like this is one of those "might not fit the current documentation" situations.

Are you intending to make it:
- ⭐ with footnote 
- ❌ with footnote
- ❔ with footnote

... or do we need a third state?

File: docs/docs/codes/_footnotes/ios-application.mdx
Comment (innovaker) [inline docs/docs/codes/_footnotes/ios-application.mdx:1]:
```suggestion
Key press is registered, but has no visible effect in tested apps (Gmail, Safari, Notes)
```

File: docs/src/data/footnotes.js
Comment (innovaker) [inline docs/src/data/footnotes.js:10]:
```suggestion
import iosApplication from "@site/docs/codes/_footnotes/ios-application.mdx";
import iosPower from "@site/docs/codes/_footnotes/ios-power.mdx";
import macosPower from "@site/docs/codes/_footnotes/macos-power.mdx";
```

File: docs/src/data/footnotes.js
Comment (innovaker) [inline docs/src/data/footnotes.js:16]:
```suggestion
  iosApplication,
  iosPower,
  macosPower,
```

**Review by innovaker — COMMENTED** [review]
Thanks @onebigdoor!

Very minor suggestions, mostly around keeping the footnotes file alphabetical as it's likely to grow organically otherwise.

**Pull request merged**
